communication/controller/src/main/java/org/openbase/jul/communication/controller/AbstractRemoteClient.java

```diff
@@ -266,24 +263,24 @@ private void internalInit(final Scope scope, final ParticipantConfig participant
         }
     }
 
-    private void initListener(final rsb.Scope scope, final ParticipantConfig participantConfig) throws CouldNotPerformException {
+    private void initSubscriber(final Scope scope, final CommunicatorConfig communicatorConfig) throws CouldNotPerformException {
         try {
-            this.listener = RSBFactoryImpl.getInstance().createSynchronizedListener(scope.concat(AbstractControllerServer.SCOPE_SUFFIX_STATUS), participantConfig);
-            this.listenerWatchDog = new WatchDog(listener, "RSBListener[" + scope.concat(AbstractControllerServer.SCOPE_SUFFIX_STATUS) + "]");
-        } catch (InstantiationException ex) {
-            throw new CouldNotPerformException("Could not create Listener on scope [" + scope + "]!", ex);
+            this.subscriber = factory.createSubscriber(ScopeProcessor.concat(scope, AbstractControllerServer.SCOPE_SUFFIX_STATUS), communicatorConfig);
+            this.subscriberWatchDog = new WatchDog(subscriber, "Subscriber[" + ScopeProcessor.generateStringRep(subscriber.getScope()) + "]");
+        } catch (CouldNotPerformException ex) {
+            throw new CouldNotPerformException("Could not create Subscriber on scope [" + scope + "]!", ex);
         }
     }
 
-    private void initRemoteServer(final rsb.Scope scope, final ParticipantConfig participantConfig) throws CouldNotPerformException {
+    private void initRemoteServer(final Scope scope, final CommunicatorConfig communicatorConfig) throws CouldNotPerformException {
         try {
-            this.remoteServer = RSBFactoryImpl.getInstance().createSynchronizedRemoteServer(scope.concat(AbstractControllerServer.SCOPE_SUFFIX_CONTROL), participantConfig);
-            this.remoteServerWatchDog = new WatchDog(remoteServer, "RSBRemoteServer[" + scope.concat(AbstractControllerServer.SCOPE_SUFFIX_CONTROL) + "]");
-            this.listenerWatchDog.addObserver(middlewareReadyObserver);
-            this.listenerWatchDog.addObserver(middlewareFailureObserver);
-            this.remoteServerWatchDog.addObserver(middlewareFailureObserver);
-        } catch (RuntimeException | InstantiationException ex) {
-            throw new CouldNotPerformException("Could not create RemoteServer on scope [" + scope + "]!", ex);
+            this.rpcClient = factory.createRPCClient(ScopeProcessor.concat(scope, AbstractControllerServer.SCOPE_SUFFIX_CONTROL), communicatorConfig);
```

Comment:
We could remove the scope suffix as the MQTT RPC implementation add `/rpc` to its initialization scope anyway.